Bug - RNMapbox Integration Broken

**Describe the bug**
Our rnmapbox integration is completely broken on native iOS and Android. This was working in the past, but I believe a recent change may have broken it. We should go back through the set up process based on the library docs and see if we did something wrong.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to andrew_testing branch, and run the simulator on iOS or Android. Navigate to the create a trip page.
iOS - map component is either empty or the entire page does not load. 

Android - on my end, builds fail with this error:

```
"FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':rnmapbox_maps:compileDebugAidl'.
> Could not resolve all task dependencies for configuration ':rnmapbox_maps:debugCompileClasspath'.
   > Could not resolve com.mapbox.maps:android:10.13.0.
     Required by:
         project :rnmapbox_maps
      > Could not resolve com.mapbox.maps:android:10.13.0.
         > Could not get resource 'https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/maps/android/10.13.0/android-10.13.0.pom'.
            > Could not GET 'https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/maps/android/10.13.0/android-10.13.0.pom'. Received status code 401 from server: Unauthorized
   > Could not resolve com.mapbox.mapboxsdk:mapbox-sdk-turf:6.11.0.
     Required by:
         project :rnmapbox_maps
      > Could not resolve com.mapbox.mapboxsdk:mapbox-sdk-turf:6.11.0.
         > Could not get resource 'https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/mapboxsdk/mapbox-sdk-turf/6.11.0/mapbox-sdk-turf-6.11.0.pom'.
            > Could not GET 'https://api.mapbox.com/downloads/v2/releases/maven/com/mapbox/mapboxsdk/mapbox-sdk-turf/6.11.0/mapbox-sdk-turf-6.11.0.pom'. Received status code 401 from server: Unauthorized"
```

**Expected behavior**
We should see our map preview from the maps/NativeMap.native.js file. 

**Screenshots**

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
 - Device: iPhone 14 
 - OS:  iOS 16.2

--- a/client/components/feed/Feed.js
+++ b/client/components/feed/Feed.js
@@ -2,7 +2,7 @@ import React, { useEffect, useState } from "react";
 import { useDispatch, useSelector } from "react-redux";
 import { Link } from "expo-router";
 import { Container, Box, Text, HStack, Stack, Switch, Button } from "native-base";
-import { StyleSheet } from "react-native";
+import { StyleSheet, FlatList } from "react-native";
 import Card from "./FeedCard";
 import DropdownComponent from "../Dropdown";
 import { theme } from "../../theme";
@@ -77,14 +77,20 @@ const Feed = ({ feedType = "public" }) => {
     }
 
     if (data.length > 0) {
-      return data.map((item) => (
-        <Link
-          key={"link-key" + item?._id}
-          href={item.type === "pack" ? "/pack/" + item?._id : "/trip/" + item?._id}
-        >
-          <Card key={item?._id} type={item.type} {...item} />
-        </Link>
-      ));
+      return (
+        <FlatList
+          data={data}
+          keyExtractor={(item) => item?._id}
+          renderItem={({ item }) => (
+            <Link
+              key={"link-key" + item?._id}
+              href={item.type === "pack" ? "/pack/" + item?._id : "/trip/" + item?._id}
+            >
+              <Card key={item?._id} type={item.type} {...item} />
+            </Link>
+          )}
+        />
+      );
     } else {
       return <Text>{ERROR_MESSAGES[feedType]}</Text>;
     }

--- a/client/components/map/MapContainer.native.js
+++ b/client/components/map/MapContainer.native.js
@@ -7,9 +7,11 @@ import NativeMap from "./NativeMap";
 export function MapContainer({shape}) {
   
   if(Platform.OS === "android" || Platform.OS === "ios"){
-    <View style={[styles.nativeContainer]}>
-      <NativeMap />
-    </View>
+    return (
+      <View style={[styles.nativeContainer]}>
+        <NativeMap />
+      </View>
+    );
   }
 
 }

--- a/client/components/pack_table/Table.js
+++ b/client/components/pack_table/Table.js
@@ -1,4 +1,4 @@
-import { StyleSheet } from "react-native";
+import { FlatList, StyleSheet } from "react-native";
 import { Table, Row, Cell, TableWrapper } from "react-native-table-component";
 import { Feather } from "@expo/vector-icons";
 import { Select, Checkbox, Box, Text, HStack, Button } from "native-base";
@@ -227,21 +227,27 @@ export const TableContainer = ({ currentPack }) => {
               ))}
               style={styles.head}
             />
-            {Object.entries(groupedData).map(([category, items]) => (
-              <>
-                <CategoryRow category={category} />
-                {items.map((item, index) => (
-                  <TableItem
-                    key={index}
-                    itemData={item}
-                    checkedItems={checkedItems}
-                    handleCheckboxChange={handleCheckboxChange}
-                    index={index}
-                    flexArr={flexArr}
+            <FlatList
+              data={Object.entries(groupedData)}
+              keyExtractor={([category, items]) => category}
+              renderItem={({ item: [category, items] }) => (
+                <>
+                  <CategoryRow category={category} />
+                  <FlatList
+                    data={items}
+                    keyExtractor={(item, index) => item._id}
+                    renderItem={({ item }) => (
+                      <TableItem
+                        itemData={item}
+                        checkedItems={checkedItems}
+                        handleCheckboxChange={handleCheckboxChange}
+                        flexArr={flexArr}
+                      />
+                    )}
                   />
-                ))}
-              </>
-            ))}
+                </>
+              )}
+            />
           </Table>
           {!foodItems.length && <Button>Add Food Item</Button>}
           {!waterItem && <Water currentPack={currentPack} />}

--- a/client/utils/mapFunctions.js
+++ b/client/utils/mapFunctions.js
@@ -1,6 +1,6 @@
 import * as Location from "expo-location";
 import * as FileSystem from 'expo-file-system';
-import { saveAs } from 'file-saver';
+// import { saveAs } from 'file-saver';
 
 
 const defaultShape = {
@@ -220,7 +220,7 @@ const handleGpxDownload = async (gpxData, filename = "trail") => {
     // Check the platform (native or web)
     if (Platform.OS === 'web') {
       const blob = new Blob([gpxData], { type: 'application/gpx+xml' });
-      saveAs(blob, `${filename}.gpx`);
+      // saveAs(blob, `${filename}.gpx`);
     } else {
       const fileUri = FileSystem.documentDirectory + `${filename}.gpx`;
       await FileSystem.writeAsStringAsync(fileUri, gpxData, {

